lib/handshake_server.ml

```diff
@@ -213,6 +213,10 @@ let answer_client_hello state (ch : client_hello) raw =
     let cciphers = ch.ciphersuites in
     assure (client_hello_valid ch) >>= fun () ->
     agreed_version config.protocol_versions ch.version >>= fun version ->
+    ( if List.mem Packet.TLS_FALLBACK_SCSV cciphers then
+        guard (version = max_protocol_version config.protocol_versions) Packet.INAPPROPRIATE_FALLBACK
+      else
+        return () ) >>= fun () ->
     agreed_cipher config.ciphers cciphers >>= fun cipher ->
```

Comment:
guard (not List.mem TLS_FALLBACK_SCSV cciphers ||
        version = max_protocol_version config.protocol_versions)
      INAPPROPRIATE_FALLBACK
?

(I could also live with an `implies` operator in `Utils`... :smile: )